astropy/io/fits/convenience.py

```diff
@@ -709,6 +710,98 @@ def info(filename, output=None, **kwargs):
+    """
+    Compare two parts of a fits file, including entire FITS files, and
+    FITS HDU objects.
+
+    Parameters
+    ----------
+    inputa : str of `HDUList`
+        The filename of a FITS file on disk, or an `HDULIST` object.
+
+    inputb : str or `HDUList`
+        The filename of a FITS file on disk, or an `HDULIST` object.
+
+    ext
+        The rest of the arguments are for extension specification if your
+        inputA and inputB are string filenames (will not work if inputA and
+        inputB are HDU objects. They are flexible and are best illustrated
+        by examples.
+
+        By extension number::
+            diff('inA.fits', 'inB.fits', 0)      # the primary HDU
+            diff('inA.fits', 'inB.fits', 2)      # the second extension
+            diff('inA.fits', 'inB.fits', ext=2)  # the second extension
+
+        By name, i.e., ``EXTNAME`` value (if unique)::
+            diff('inA.fits', 'inB.fits', 'sci')
+            diff('inA.fits', 'inB.fits', extname='sci')  # equivalent
+
+        Note ``EXTNAME`` values are not case sensitive
+
+        By combination of ``EXTNAME`` and EXTVER`` as separate
+        arguments or as a tuple::
+            diff('inA.fits', 'inB.fits', 'sci', 2)  # EXTNAME='SCI' & EXTVER=2
+            diff('inA.fits', 'inB.fits', extname='sci', extver=2)  # equivalent
+            diff('inA.fits', 'inB.fits', ('sci', 2))  # equivalent
+
+        Ambiguous or conflicting specifications will raise an exception::
+            diff('inA.fits', 'inB.fits', ext=('sci',1), extname='err', extver=2)
+
+
+    kwargs
+        Any additional keyword arguments to be passed to
+        `astropy.io.fits.FITSDiff`.
+
+    Notes
+    -----
+    The primary use for the `diff` function is to allow quick print out
+    of a FITS difference report and will write to ``sys.stdout``.
+    To save the diff report to a file please use `astropy.io.fits.FITSDiff`
+    directly.
+    """
+
+    if isinstance(inputa, string_types) and isinstance(inputb, string_types):
+
+        # Use ext keywords if filename strings provided, unless none exist, then run FITSDIFF directly
+        if len(args) == 0 and 'ext' not in kwargs and 'extname' not in kwargs and 'extver' not in kwargs:
+            print(FITSDiff(inputa, inputb, **kwargs).report())
+
+        else:
+            # Use handy _getext to interpret any ext keywords
+            modea, closeda = _get_file_mode(inputa)
+            modeb, closedb = _get_file_mode(inputb)
+            hdulista, extidxa = _getext(inputa, modea, *args, **kwargs)
+            hdulistb, extidxb = _getext(inputa, modeb, *args, **kwargs)
+
+            # Pop those extension keywords if they exist
+            kwargs.pop('ext', None)
+            kwargs.pop('extname', None)
+            kwargs.pop('extver', None)
+
+            try:
+                hdua = hdulista[extidxa]
+                hdub = hdulistb[extidxb]
+                print(HDUDiff(hdua, hdub, **kwargs).report())
```

Comment:
OK, seems fair enough now that I reread the original issue. Could you maybe add a comment in the code so no one doing a general cleanup runs into this same argument?